### PR TITLE
Gracefully handle EC diverging from F3

### PR DESCRIPTION
**Pull request by Kubuxu**
We will end up proposing base which will trigger backoff, giving us chance to re-boostrap,
Also added a metric for it.

### CODE REVIEW

**Comment by Kubuxu** [comment]
For some reason tests fail with what looks like OOM.

File: ec/ec.go [hunk 1]
```diff
@@ -26,4 +26,7 @@ type TipSet interface {
 	Beacon() []byte
 	Epoch() int64
 	Timestamp() time.Time
+
+	// User readable name of the tipset
+	String() string
```
Comment (Stebalien) [inline ec/ec.go:31]: nit: put `fmt.Stringer` at the top.

File: ec/fake_ec.go [hunk 2]
```diff
@@ -54,6 +55,14 @@ func (ts *Tipset) Timestamp() time.Time {
 	return ts.timestamp
 }
 
+func (ts *Tipset) String() string {
+	res := base64.RawStdEncoding.EncodeToString(ts.tsk[:gpbft.CidMaxLen])
```
Comment (Stebalien) [inline ec/fake_ec.go:59]: I'd check the length before slicing.

Comment (Stebalien) [reply ec/fake_ec.go:59]: Honestly, if we're going to stringify the tipset, I'd prefer to just stringify it as proper CIDs.

File: host.go [hunk 3]
```diff
@@ -361,20 +361,20 @@ func (h *gpbftHost) collectChain(base ec.TipSet, head ec.TipSet) ([]ec.TipSet, e
...
-			// we reorged away from base
-			// scream and panic??
-			// TODO make sure this is correct, re-boostrap/manifest swap code has to be able to
-			// catch it
-			panic("reorg-ed away from base, dunno what to do, reboostrap is the answer")
+	current := head
+	for !bytes.Equal(current.Key(), base.Key()) {
+		if current.Epoch() < base.Epoch() {
+			metrics.headDiverged.Add(h.runningCtx, 1)
+			log.Errorw("reorg-ed away from base, reboostrap is the answer",
```
Comment (Stebalien) [inline host.go:368]: This is going to _majorly_ annoy users.

Comment (Kubuxu) [reply host.go:368]: It will happen only every minute or so, but agreed, I will downgrade to Warn and tone it down

File: ec/fake_ec.go (hunk 2 above)
Comment (Kubuxu) [reply ec/fake_ec.go:59]: If the length is wrong, then it means someone messed with it. The FakeEC will always create proper length, the test is there so if anyone changes it, it gets detected.

I'm importing multibase, at this point we might import go-cid but I'm not touching that right now.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**